airflow/decorators/base.py

```diff
@@ -351,7 +351,7 @@ def fake():
         except TypeError:  # Can't evaluate return type.
             return False
         ttype = getattr(return_type, "__origin__", return_type)
-        return ttype is dict or ttype is Dict
+        return issubclass(ttype, Dict)
```

Comment:
I am not 100% a type checking expert. A dictionary is also just marked by it is "mappable" and "Mapping" is the abstract check for this. I agree that the previous check was not good but I don't think `issubclass()` is also covering all use cases.
```suggestion
        return isinstance(ttype, Mapping)
```
See also: https://docs.python.org/3/library/collections.abc.html#collections-abstract-base-classes

Reply:
I agree, `typing.Mapping` is more suitable. I did not want to change the code drastically,
and since we are comparing types here, and not instances, we should still use `issubclass()` and not `isinstance()`.